solrsearch-plugins/src/main/java/co.cask.hydrator.plugin/common/SolrSearchSinkConfig.java

```diff
@@ -0,0 +1,221 @@
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Config class for SolrSearchSink and RealtimeSolrSearchSink plugins.
+ */
+public class SolrSearchSinkConfig extends ReferencePluginConfig {
+  public static final String SINGLE_NODE_MODE = "SingleNode";
+  public static final String SOLR_CLOUD_MODE = "SolrCloud";
+
+  @Description("Solr mode to connect to. For example, Single Node Solr or SolrCloud.")
+  private final String solrMode;
+  @Description("The hostname and port for the Solr server separated by a colon. For example, localhost:8983 for " +
+    "Single Node Solr or comma-separated list of hostname and port, zkHost1:2181,zkHost2:2181,zkHost3:2181 " +
+    "for SolrCloud.")
+  private final String solrHost;
+  @Description("Name of the collection where data will be indexed and stored in Solr.")
+  private final String collectionName;
+  @Description("Field that will determine the unique key for the document to be indexed. It must match a " +
+    "field name in the structured record of the input.")
+  private final String keyField;
+  @Description("List of the input fields to map to the output Solr fields. The key specifies the name of the field to" +
+    " rename, with its corresponding value specifying the new name for that field.")
+  @Nullable
+  private final String outputFieldMappings;
+
+  public SolrSearchSinkConfig(String referenceName, String solrMode, String solrHost, String collectionName,
+                              String keyField, String outputFieldMappings) {
+    super(referenceName);
+    this.solrMode = solrMode;
+    this.solrHost = solrHost;
+    this.collectionName = collectionName;
+    this.keyField = keyField;
+    this.outputFieldMappings = outputFieldMappings;
+  }
+
+  /**
+   * Returns the Solr mode selected as input by user.
+   *
+   * @return Solr mode
+   */
+  public String getSolrMode() {
+    return solrMode;
+  }
+
+  /**
+   * Returns the hostname and port entered by user.
+   *
+   * @return Solr host
+   */
+  public String getSolrHost() {
+    return solrHost;
+  }
+
+  /**
+   * Returns the Solr collection name given as input by user.
+   *
+   * @return collection name
+   */
+  public String getCollectionName() {
+    return collectionName;
+  }
+
+  /**
+   * Returns the Key field given as input by user.
+   *
+   * @return key field
+   */
+  public String getKeyField() {
+    return keyField;
+  }
+
+  /**
+   * Returns the input fields and its output fields mapping given as input by user.
+   *
+   * @return
+   */
+  @Nullable
+  public String getOutputFieldMappings() {
+    return outputFieldMappings;
+  }
+
+  /**
+   * Creates a map for input field and its mapping for output field, if present.
+   *
+   * @return Map of input fields and its mapping
+   */
+  public Map<String, String> getOutputFieldMap() {
+    Map<String, String> outputFieldMap = new HashMap<String, String>();
```

Comment:
this method is called often, so every time this is called, you create a  map, parse, add to map and return. very inefficient. parse and store the map in a local variable and return that variable if non-null for subsequent calls.